### PR TITLE
logic

### PR DESCRIPTION
**Pull request by LentonGar**

### CODE REVIEW

**Comment by LentonGar** [comment]
sharing first attempts at logic that work in two variations - point-radius, opening hours; and polygon-point, opening hours.

**Review by kislerdm — COMMENTED** [review]
Hey @LentonGar! Thank you for defining the filtering logic. Could you please review the comments and consider adjustments accordingly?

Also, WDYT about benchmarking? You proposed two ways to apply geolocation filter - which one shall be used in the app? If both, what shall be the criteria of which one to use when?

**(issue)** Please leave a concise description of the codebase change when opening PR - it sets expectations for reviewers.

Thanks!

File: Deft_logic_with_polygon.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    name: str
+    email: str
+    coordinates: Tuple[float, float]
+    opening_hours: 'Business_hours'
+
+@dataclasses.dataclass
+class LostRequest:
+    description: str
+    email: str
+    polygon: List[Tuple[float, float]]  # List of coordinates representing the polygon
```
[the remark this answers is not in the document]
Comment (LentonGar) [reply Deft_logic_with_polygon.py:16]: wouldn't that exclude everywhere west of London and east of New Zealand? I think it's safe to assume we don't need to cover the north and south poles, so -90 -> 90 is good enough for latitude. 

```python
class Point:
    def __init__(self, lon: float, lat: float):
        if lon <= -180 or lon >= 180:
            raise ValueError("longitude must be between -179 and 180 degrees")
        if lat < -90 or lat > 90:
            raise ValueError("latitude must be between -90 and 90 degrees")
        
        self.lon = lon
        self.lat = lat

    def __eq__(self, other: "Point") -> bool:
        return self.lon == other.lon and self.lat == other.lat
```


```suggestion
polygon: List[Point]
```

**Comment by LentonGar** [comment]
Thank you for the feedback, Dmitry!

Re. which one is to be used in the app, I wasn't certain, which is why I tried to do both. I suppose polygon-point is more useful, but I don't how we will obtain those inputs yet. Point-radius is easier to obtain but less useful. I'll bring it up today in the meeting. 

Re. GH conventions, I will try to my best to be as conventional as possible by the time we finish (from now on). I am still very much fumbling around but appreciate that learning to use and navigate Git(hub) is as much a part of the learning process as anything else. Thank you for pointing me in the right direction in that sense!

**Comment by kislerdm** [comment]
I'd suggest to close PR given it's stale. The takeaway: keep track of PRs with the "time-to-first comment" to be less than 24 hours.

**Pull request closed without merging**